diff --git a/bin/git-fuzzy b/bin/git-fuzzy
index 27eaa4f..a33dc8c 100755
--- a/bin/git-fuzzy
+++ b/bin/git-fuzzy
@@ -21,6 +21,7 @@ script_dir="$( cd -P "$( dirname "$script_source" )" >/dev/null 2>&1 && pwd )"
 # -----------------------------------------
 git_fuzzy_dir="$script_dir/.."
 
+. "$git_fuzzy_dir/lib/load-configs-from-git.sh"
 . "$git_fuzzy_dir/lib/load-configs.sh"
 
 . "$git_fuzzy_dir/lib/snapshot.sh"
diff --git a/lib/load-configs-from-git.sh b/lib/load-configs-from-git.sh
new file mode 100644
index 0000000..58b6371
--- /dev/null
+++ b/lib/load-configs-from-git.sh
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+# <<AUTOMATICALLY GENERATED>>
